exec/cnex/cnex.c

```diff
@@ -315,6 +370,50 @@ TExecutor *exec_newExecutor(TModule *object)
     return r;
 }
 
+int findDebugLine(TModule *m, int ip)
+{
+    cJSON *pStart = m->debug_symbols;
+    cJSON *jip = NULL;
+    cJSON *jln = NULL;
+    cJSON *jelm = NULL;
+    cJSON *lines = NULL;
+    // Iterate the line_numbers JSON element, looking for the source code line that is 
+    // associated with the provided Instruction Pointer (ip) value.
+    lines = cJSON_GetObjectItem(m->debug_symbols, "line_numbers");
+    cJSON_ArrayForEach(jelm, lines)
+    {
+        jip = cJSON_GetArrayItem(jelm, 0);
+        if (!cJSON_IsInvalid(jip) && jip->valueint == ip) {
+            // We found the line, so break out, and return it.
+            jln = cJSON_GetArrayItem(jelm, 1);
+            break;
+        }
+    }
+    // Reset the debug pointer back to the start of the JSON object.
+    m->debug_symbols = pStart;
```

Comment:
This may have been a fundamental misunderstanding on my part, in relation to how cJSON parses JSON data.  I just validated that I do not need to be resetting this pointer. -- Fixed.